### PR TITLE
Fix transaction sandblast filter preventing trees from being updated

### PR DESCRIPTION
**Pull request by AloeareV**

### CODE REVIEW

**Comment by AloeareV** [comment]
Fixes #483 and #484

File: zingolib/src/blaze/trial_decryptions.rs [hunk 1]
```diff
@@ -167,26 +169,59 @@ impl TrialDecryptions {
...
+                        )
+                    })
+                    .collect();
+
                 if let Some(filter) = transaction_size_filter {
                     if compact_transaction.outputs.len() + compact_transaction.actions.len()
                         > filter as usize
                     {
-                        break;
+                        sapling_notes_to_mark_position_in_block
```
Comment (fluidvanadium) [inline zingolib/src/blaze/trial_decryptions.rs:207]: we use .extend_from_slice here... and again on line 237??

**Review by fluidvanadium — COMMENTED** [review]
why do we add orchard_notes_to_mark_position_in_tx to orchard_notes_to_mark_position twice?
is mark position the same as add witness mark?

File: zingolib/src/blaze/trial_decryptions.rs (hunk 1 above)
Comment (AloeareV) [reply zingolib/src/blaze/trial_decryptions.rs:207]: We call `continue` on the next line, which skips to the next iteration of the loop

Comment (fluidvanadium) [reply zingolib/src/blaze/trial_decryptions.rs:207]: aha.

Comment (fluidvanadium) [reply zingolib/src/blaze/trial_decryptions.rs:207]: so im reading if the transaction is bigger than the filter...

Comment (fluidvanadium) [reply zingolib/src/blaze/trial_decryptions.rs:207]: then we SKIP trial decryption, BUT we STILL mark_position of every note in it as ephemeral?

**Review by fluidvanadium — APPROVED** [review]
alright she know what it do

**Pull request merged**